packages/contracts/test/withdrawer.spec.ts

```diff
@@ -0,0 +1,213 @@
+import { expect } from 'chai'
+import { Signer, BigNumber, Wallet, providers, ethers, utils } from 'ethers'
+import {
+  Withdrawer,
+  Withdrawer__factory,
+  TestLibSecureMerkleTrie,
+  TestLibSecureMerkleTrie__factory,
+  WithdrawalVerifier,
+  WithdrawalVerifier__factory,
+  L2OutputOracle,
+  L2OutputOracle__factory,
+} from '../typechain'
+import { toRpcHexString, toHexString, Address } from '@eth-optimism/core-utils'
+import * as rlp from 'rlp'
+
+const l2GethProvider = new providers.JsonRpcProvider('http://localhost:9545')
+const l1GethProvider = new providers.JsonRpcProvider('http://localhost:8545')
+
+const withdrawerAddress = '0x4200000000000000000000000000000000000015'
+
+const NON_ZERO_ADDRESS = '0x' + '11'.repeat(20)
+const NON_ZERO_GASLIMIT = BigNumber.from(50_000)
+const NON_ZERO_DATA = '0x' + '11'.repeat(42)
+
+if (!process.env.PRIVATE_KEY) {
+  throw new Error('You must define PRIVATE_KEY in your environment.')
+}
+
+const encodeWithdrawal = (args: {
```

Comment:
Longer term, we should make a `class Withdrawal` that has `encode` and `decode` methods on it